tests/unit/src/unitstd/EReg.unit.hx

```diff
@@ -120,7 +120,7 @@ new EReg("^" + EReg.escape("\\ ^ $ * + ? . ( ) | { } [ ]") + "$", "").match("\\
 ~/(\d+)/g.replace("a1234b12","$$1") == "a$1b$1";
 
 // #10592 - null character
-#if (!hl && !php)
+#if (!hl && (!php || php_ver >= 8.2))
```

Comment:
We could use also use: https://www.php.net/manual/en/function.phpversion.php

Reply:
I'm not sure that would play well with how our unit tests work

Reply:
We have an if statement like that in the Date tests: https://github.com/HaxeFoundation/haxe/blob/2388b89e93a43d888ee7f69dafd707e85b311f23/tests/unit/src/unitstd/Date.unit.hx#L159